Sources/AWSLambdaRuntimeCore/LambdaHandler.swift

```diff
@@ -44,45 +46,129 @@ public protocol LambdaHandler: EventLoopLambdaHandler {
     ///
     /// - Returns: A Lambda result ot type `Output`.
     func handle(_ event: Event, context: LambdaContext) async throws -> Output
+
+    /// Encode a response of type ``Output`` to `ByteBuffer`.
+    /// Concrete Lambda handlers implement this method to provide coding functionality.
+    /// - parameters:
+    ///     - value: Response of type ``Output``.
+    ///     - buffer: A `ByteBuffer` to encode into, will be overwritten.
+    ///
+    /// - Returns: A `ByteBuffer` with the encoded version of the `value`.
+    func encode(value: Output, into buffer: inout ByteBuffer) throws
+
+    /// Decode a `ByteBuffer` to a request or event of type ``Event``.
+    /// Concrete Lambda handlers implement this method to provide coding functionality.
+    ///
+    /// - parameters:
+    ///     - buffer: The `ByteBuffer` to decode.
+    ///
+    /// - Returns: A request or event of type ``Event``.
+    func decode(buffer: ByteBuffer) throws -> Event
 }
 
 @available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
 extension LambdaHandler {
-    public static func makeHandler(context: LambdaInitializationContext) -> EventLoopFuture<Self> {
-        let promise = context.eventLoop.makePromise(of: Self.self)
+    public init() {
+        self.init()
+    }
```

Comment:
This looks very much like infinite recursion to me:

```swift
import XCTest
import AWSLambdaRuntime

struct Foo: Codable {}
struct Bar: Codable {}

struct FooHandler: LambdaHandler {
    typealias Event = Foo
    typealias Output = Bar

    var haha: String

    init(context: LambdaInitializationContext) async throws {
        self.haha = "hahaha"
    }

    func handle(_ event: Foo, context: LambdaContext) async throws -> Bar {
        preconditionFailure()
    }
}

@available(macOS 12, *)
final class FooTests: XCTestCase {

    func testInitialization() {
        let handler = FooHandler()

        print(handler)
    }
}
```

This compiles but crashes then.

Reply:
infinite recursion.thanks was also my exception but the tests in `LambdaHandlerTest` work fine. specifically see testBootstrapSuccess. testBootstrapFailure, testBootstrapInitSimple, testBootstrapNoInit. thoughts?

Reply:
added `SimpleLambdaHandler` back